src/Agent.Sdk/ProcessInvoker.cs

```diff
@@ -244,14 +244,16 @@ public ProcessInvoker(ITraceWriter trace)
                 Interlocked.Increment(ref _asyncStreamReaderCount);
             }
 
-#if OS_WINDOWS
             // If StandardErrorEncoding or StandardOutputEncoding is not specified the on the
             // ProcessStartInfo object, then .NET PInvokes to resolve the default console output
             // code page:
             //      [DllImport("api-ms-win-core-console-l1-1-0.dll", SetLastError = true)]
             //      public extern static uint GetConsoleOutputCP();
-            StringUtil.EnsureRegisterEncodings();
-#endif
+            if (PlatformUtil.RunningOnOS == PlatformUtil.OS.Windows)
```

Comment:
It would be nice to have functions like `IsRunningOnWindows()` and `IsRunningOnLinux()` to make this read a bit easier.

Reply:
Agreed. I think I'll put that in a separate PR.